lib/core/utils/get-selector.js

```diff
@@ -1,104 +1,218 @@
-/**
- * Gets the index of element siblings that have the same nodeName
- * Intended for use with the CSS psuedo-class `:nth-of-type()` and xpath node index
- * @param  {HTMLElement} element The element to test
- * @return {Number}         The number of preceeding siblings with the same nodeName
- * @private
- */
-function nthOfType(element) {
-	'use strict';
-
-	var index = 1,
-		type = element.nodeName.toUpperCase();
-
-	/*jshint boss:true */
-	element = element.previousElementSibling;
-	while (element) {
-		if (element.nodeName.toUpperCase() === type) {
-			index++;
-		}
-		element = element.previousElementSibling;
-	}
-
-	return index;
+const escapeSelector = axe.utils.escapeSelector;
+
+function isUncommonClassName (className) {
```

Comment:
I **think** this will yield better results. I was thinking of things like the "col-md-12" stuff from bootstrap, or "sr-only". Those might be used in different places on the page. It seemed to me that having that info is better than having the nodeName. It massively depends ofcourse.